google_guest_agent/osinfo/osinfo_windows.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2023 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osinfo
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+)
+
+var (
+	version                     = windows.NewLazySystemDLL("version.dll")
```

Comment:
Can you document these variables? giving a bit more context to what they mean.

Reply:
Added the links to these procedure calls in msft docs, that seems pretty self explanatory?

Reply:
I wanted the code to be self-contained - to not depend on external resources/links. But we can neglect it for now.